wagtail/wagtailcore/blocks/base.py

```diff
@@ -204,24 +207,100 @@ def get_prep_value(self, value):
         return value
 
     def get_context(self, value):
+        """
+        Return a dict of context variables (derived from the block value, or otherwise)
+        to be added to the template context when rendering this value through a template.
+        """
         return {
             'self': value,
             self.TEMPLATE_VAR: value,
         }
 
-    def render(self, value):
+    def _render_with_context(self, value, context=None):
+        """
+        Temporary hack to accommodate Block subclasses created for Wagtail <1.6 which
+        have overridden `render` with the type signature:
+            def render(self, value)
+        and will therefore fail when passed a `context` kwarg.
+
+        In Wagtail 1.8, when support for context-less `render` methods is dropped,
+        this method will be deleted (and calls to it replaced with a direct call to `render`).
+        """
+        try:
+            return self.render(value, context=context)
+        except TypeError:
```

Comment:
Yes, I haven't expected to see `Exception: Sad trombone sound` and it took some time to figure out why I've got this exception instead of `TypeError` that was thrown by the `paginator`.

`inspect` - it's just the first idea that came to my head. If you know more Pythonic way, feel free to use it ;)